.github/workflows/check_smoke_tests.yml

```diff
@@ -0,0 +1,96 @@
+name: check smoke tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  smoke_tests:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out repository
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Set up Docker
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y docker-compose
+
+    - name: Start services with docker-compose
+      run: docker-compose up -d
```

Comment:
The `compose.yml` file is currently configured for development use, the main differences being it will run jaeger by default and override the default CMD to wrap the web server in an opentelemetry call.  Neither of these will likely be used in production environments.  Did you consider just using docker on its own to run the API?  The only dependency would be a database like postgres.

Reply:
Yeah I could definitly change it to use that. No real reason I did it this way other than it was the first thing I thought of. I'll take a look at just using the dockerfile instead